src/ol/layer/STAC.js

```diff
@@ -466,6 +470,25 @@ class STACLayer extends LayerGroup {
 
     const sources = [];
     switch (link.rel) {
+      case 'pmtiles':
+        const p = new pmtiles.PMTiles(link.href);
```

Comment:
Make sure to pass headers and query parameters for Auth, see https://github.com/protomaps/PMTiles/pull/112

Reply:
That PR doesn't accomplish 100% of the functionality required for this - it's more straightforward to write a custom `Source` implementation

Reply:
@m-mohr actually it might be enough, is there a well-trodden use case here - just an `Authentication` header or also query params?

Reply:
Yeah, it's a general issue in the OL STAC and stac-layer implementations. I need to check this more in details together with STAC Browser, where this occurs.

The use case is authentication, which is usually appending an arbitrary query parameter to the URL (e.g. an API token) or an arbitrary HTTP header (e.g. a Bearer Token). It would be good if there would be a hook where e.g. STAC Browser could intercept and modifiy the request according to the needs of the deployment.

Maybe I just take this over once I'm fully implementing this in STAC Browser and OL STAC and make PRs to the affected libraries, including ol-pmtiles.